### DISCUSSION TITLE
Geometries not updating as zoom moves - locked bounds

### DISCUSSION THREAD

**Discussion by jjcfrancisco** [maplibre/martin#1061] [Q&A]
Hi all,

Our cartographers have noticed that in `martin` `v0.11.3` features aren't fully loading until a certain zoom is hit - such zoom is not matching the viewport of the window. I want to believe this is not connected to our data or our martin configs. For instance, I am simply swapping images: when using `v0.10.0` I get no issues but when using anything above it (i.e. `latest`) features get temporarily 'locked' into a specific bound. Hopefully the pictures make sense. This may be a bug but I don't want to speculate and so sharing it as a discussion.

`v0.11.3`: features getting *chopped*
![without](https://github.com/maplibre/martin/assets/49205177/1369f951-7bc6-493a-ad85-1a54563d4d36)

`v0.11.3`: superposed red lines to highlight problem
![with-lines](https://github.com/maplibre/martin/assets/49205177/bbebbe40-8b54-4a60-a4d0-425b1c30df60)

`v0.10.0`: features aren't *chopped* anymore and displaying geometries as they should
![fine](https://github.com/maplibre/martin/assets/49205177/4db21570-b992-4b36-a7e3-ce871f83086b)

Thanks

**Comment by nyurik** [comment]
@jjcfrancisco thanks for reporting, this is an interesting issue.  Could you try running martin with the env var `RUST_LOG=debug` (or even `trace`, but that might be too much) ?  (BTW, we really ought to write a guide on debugging Martin).  You should see what query is being made into the database - and compare if the query is the same between the versions.  Martin creates an SQL query and sends it to the DB. The query should not (in theory) have changed, but everything is possible.

Other than that, other issues might be:
* feature count - if you limit the number of features per tile, and you have a high-enough feature count, some might be dropped
* tile buffer miscalculation (someone recently reported there is a problem with certain projections and buffering growing incorrectly near date line, but i doubt this is the case)
* ???

**Reply by jjcfrancisco** [reply]
Hi @nyurik, thanks for the reply. I ran martin with both log levels but I didn't see anything unusual - bear in mind that my knowledge of rust is small. I get two SQL queries in the stdout that I guess is from martin inspecting the database (i.e. `pg_catalog` / `information_schema`). But we use custom functions with auto_publish set to `false` and I couldn't see the custom function in the stdout, I did notice calls to the function when the map is loaded. Feature count isn't set in our configs but documentation says that default is `unlimited` so that shouldn't be an issue. In terms of SRID we just use the web mercator... hopefully that's not an issue either?

Again, to me what seems strange is that without changing anything (i.e. no configs changed) the results are different when changing images. Here are the versions I tested:
- `v0.9.3` -> no chopping
- `v0.10.0` -> no chopping
- `v0.11.0` -> chopping occurs
- `v0.11.1` -> chopping occurs
- `v0.11.2` -> chopping occurs
- `v0.11.3` -> chopping occurs

Let me know if there's something more specific I could do.

**Reply by nyurik** [reply]
@jjcfrancisco thx for the info. I agree that this should not be happening, but I do need some help to track it down. You do not need to have any Rust experience at all. You need to run Martin with `debug` log level, and ignore all of the initial startup logging - that stuff is not important. Once the server is fully up, use `curl` or browse to it to force it to actually produce tiles. During that time, you should see some SQL statements in the logs. That's all i am interested in -- try to find any SQL statements that have `ST_AsMVT` in them -- those are the ones producing the actual tiles.  Compare the SQL with the older Martin version.  Message me on Slack if you have some non-public info that you don't want to disclose.

**Comment by IncidGeo** [comment]
I've encountered the same issue, but if i restart Martin the problem disappear for a certain time then come again. In my case it does appear after writing new geometries in the database.

**Reply by IncidGeo** [reply]
I'm running Martin v0.8.2 and i still have geometries "chopped"

**Reply by jjcfrancisco** [reply]
@IncidGeo I am in `v0.11.6` and have no chopping issues anymore. It got fixed in `v0.11.5` (see [here](https://github.com/maplibre/martin/releases/tag/v0.11.5)). Are you able to move up in versions?

**Reply by IncidGeo** [reply]
Good news ! Yes i can change of version but as my project test is running on Arm / Docker i'll need to create my own image. But maybe the actual image is now compatible with all platforms.

**Reply by nyurik** [reply]
I have included docker arm images, see if they work for you

**Reply by jjcfrancisco** [reply]
I've used both images in arm64 + amd64 and they worked fine for `v0.11.6` :)

**Reply by IncidGeo** [reply]
I've upgrade to v0.13 docker image it has solved my slow start up issue, but i still have the disappear/chopped geometries problem.
v0.11.6 seems ok for startup time up and geometries
